### DISCUSSION TITLE
Zen browser coapp detection

### DISCUSSION THREAD

**Discussion by namaelsan** [aclap-dev/video-downloadhelper#1339] [Help and Q&A]
### Your question

Im using zen browser as a flatpak. I want to use vdh but even though i have the coapp installed, the extension doesnt detect it. I saw that the installer did some tweaks to the brave, firefox etc. flatpaks, i tried to dig them up myself with flatseal but the only change i saw there was that the ~/.local/share/vdhcoapp:ro option was added to the filesystems. Even though i did the same to zen browser, the extension still cant detect the coapp. So i was wondering if i could learn what those tweaks were so i could apply them myself, some help would be appreciated :)
edit:
i also ran the command "flatpak permission-set webextensions net.downloadhelper.coapp io.github.zen_browser.zen yes" but the outcome is the same

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":false,"error":"No such native application net.downloadhelper.coapp"}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 2
}

**Comment by ddpasa** [comment]
Also having the same issue, but coapp detection works fine on Chromium.

**Comment by makissuelms** [comment]
I'm having the same problem

**Comment by Mattmjs** [comment]
Refer to the wiki: https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems

If this doesn't help, maybe ask the people who make this browser. It may not be a supported browser.

**Reply by ddpasa** [reply]
> Refer to the wiki: https://github.com/aclap-dev/video-downloadhelper/wiki/CoApp-not-recognized#ubuntu-and-flatpak--snap-based-systems
> 
> If this doesn't help, maybe ask the people who make this browser. It may not be a supported browser.

Zen is just a slightly modified version of Firefox. Normal Firefox plugins work fine there. This looks like a flatpak issue.

**Reply by julianzier** [reply]
Any news? Having the same problem.

**Comment by pawatecnologia** [comment]
I'm also having the same problem. I tried copying the `native-messaging-hosts/net.downloadhelper.coapp.json` folder to `~/.var/app/app.zen_browser.zen/`, but it still didn't work.

I have the midorian browser. And it works with coapp without any additional configuration.

**Comment by Mattmjs** [comment]
Maybe it's a problem with the flatpack app, I tested out  zen browser as an appimage and as a tar package.
I tested this on Linux Mint in a virtual machine where Firefox is already installed. This is important because the installation of 
the companion app writes a native messaging hosts folder in the .mozilla directory. The software assumes you're using one of
the 3 officially supported browsers, Firefox, Chrome and Edge. Zen along with other Firefox forks are not officially supported 
browsers.

Both package formats work and the extension can see the installed companion app. The interesting part is that zen somehow 
uses the native messaging from .mozilla directory. If you plan on purging Firefox from Linux afterwards then copy the native messaging
hosts folder from .mozilla and paste in .zen, see the image below running as an appimage

![Zen browser app image](https://github.com/user-attachments/assets/0f0df523-ab19-4640-9262-f0791d392b69)

**Reply by guilhermeakiow** [reply]
Step by step, please?

**Reply by Mattmjs** [reply]
What do you not understand ?

**Reply by guilhermeakiow** [reply]
I didn't understand the step by step instructions on how you installed it on Zen. I'm using macOS intel.

**Reply by Mattmjs** [reply]
Yes I did my testing on Linux, you can get it working from here #216 
If you don't have Firefox installed then you'll need to do that , then reinstall the companion app.
This is to copy the native messaging hosts over to Zen from Mozilla directory.

**Reply by Greqos** [reply]
Whats helped me to get it work was opening Flatseal and allowing Zen Browser to access ~/.mozilla directory. It looks like Zen Browser reads the native messaging directly from .mozilla directory in home folder, no need to copy it to Zen data folder.

![Zrzut ekranu z 2025-05-05 23-23-22](https://github.com/user-attachments/assets/a55cc28e-6600-46d5-b710-ecc6599ffae6)

**Reply by nataondev** [reply]
It's work!

**Reply by rexendevar** [reply]
Doesn't work for me. Tried granting it permissions to .mozilla and copying over the file and all that. Doesn't do anythning.

**Reply by Greqos** [reply]
First check if it works on Firefox. If not the problem is elsewhere.

**Reply by talison-cardoso** [reply]
It worked here.
You need to add all three paths. Not just the one marked in the photo.
`~/.mozilla`
`/opt/vdhcoapp:ro`
`/home/USER/.local/share/vdhcoapp:ro`

**Reply by Hanibachi** [reply]
worked for me as well, no need for /otp/vdhcoapp:ro
